Classes/Emogrifier.php

```diff
@@ -530,6 +530,13 @@ private function fillStyleAttributesWithMergedStyles()
     private function generateStyleStringFromDeclarationsArrays(array $oldStyles, array $newStyles)
     {
         $combinedStyles = array_merge($oldStyles, $newStyles);
+
+        foreach ($oldStyles as $attribute => $expression) {
+            if (isset($newStyles[$attribute]) && strtolower(substr($expression, -10)) === '!important') {
```

Comment:
Then we'll probably need to either use a regular expression, or we'd need another trim (or make sure that the elements in $oldStyles are trimmed when setting).